File: py_payment_api/app/resources/store.py
from flask_restful import Resource
from ..services import payment_service, apple_subscription_service, coinbase_subscription_service, google_subscription_service

class StoreItems(Resource):
    """
    Resource class to handle store item retrieval.
    """

    def get(self):
        """
        Handles GET requests to retrieve store items.

        Returns
        -------
        dict
            A dictionary containing the list of store items.
        """
        
        result = {}
        result["items"] = payment_service.get_items()
        result["subscriptions"] = {
            "apple": apple_subscription_service.get_items(),
            "coinbase": coinbase_subscription_service.get_items(),
            "google": google_subscription_service.get_items()
        }
        return result, 200

File: tsPaymentApi/src/resources/store.ts
import { Request, Response } from 'express';
import { paymentService, appleSubscriptionService, coinbaseSubscriptionService, googleSubscriptionService } from '../services';

export class StoreItems {
    get(req: Request, res: Response): void {
        const result: any = {};
        result.items = paymentService.getItems();
        result.subscriptions = {
            apple: appleSubscriptionService.getItems(),
            coinbase: coinbaseSubscriptionService.getItems(),
            google: googleSubscriptionService.getItems()
        };
        res.status(200).json(result);
    }
} 